### PR TITLE
integrate swagger

### PR DESCRIPTION
**Pull request by astha-singh-2445**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,4 +1,4 @@
-plugins {
+ plugins {
```
Comment (ShivamNagpal) [inline build.gradle:1]: nit: Extra whitespace

File: src/main/java/com/singh/astha/notificationservice/config/AppConfig.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+@Configuration
+@OpenAPIDefinition(info = @Info(title = "My API", version = "v1"))
+@SecurityScheme(
+        name = "bearerAuth",
+        type = SecuritySchemeType.HTTP,
+        bearerFormat = "JWT",
+        scheme = "bearer"
+)
+public class AppConfig {
```
Comment (ShivamNagpal) [inline src/main/java/com/singh/astha/notificationservice/config/AppConfig.java:18]: nit: If this file is just for the Swagger Docs configuration, then we can name it OpenAPIConfig

File: src/main/java/com/singh/astha/notificationservice/config/SecurityConfig.java [hunk 3]
```diff
@@ -19,7 +19,7 @@ public SecurityConfig(JwtFilter jwtFilter, JwtAuthenticationHandler jwtAuthentic
 
     @Override
     protected void configure(HttpSecurity http) throws Exception {
-        http.csrf().disable().authorizeRequests().antMatchers("/health").permitAll().anyRequest().authenticated().and()
+        http.csrf().disable().authorizeRequests().antMatchers("/health","/api-docs/**","/swagger-ui/**").permitAll().anyRequest().authenticated().and()
```
Comment (ShivamNagpal) [inline src/main/java/com/singh/astha/notificationservice/config/SecurityConfig.java:22]: File is not formatted

File: src/main/resources/application.properties [hunk 4]
```diff
@@ -3,4 +3,7 @@ spring.data.mongodb.auto-index-creation=true
 server.port=${SERVER_PORT:8081}
 spring.kafka.consumer.group-id=astha
 auth.token.public.key=${PUBLIC_KEY}
-spring.kafka.bootstrap-servers=${KAFKA_BOOTSTRAP_SERVERS}
\ No newline at end of file
+spring.kafka.bootstrap-servers=${KAFKA_BOOTSTRAP_SERVERS}
+
+springdoc.api-docs.path=/api-docs
+springdoc.swagger-ui.path=/swagger-ui-custom.html
```
Comment (ShivamNagpal) [inline src/main/resources/application.properties:9]: nit: Put a blank line at the end of file

File: build.gradle (hunk 1 above)
Comment (astha-singh-2445) [reply build.gradle]: fixed

File: src/main/java/com/singh/astha/notificationservice/config/AppConfig.java (hunk 2 above)
Comment (astha-singh-2445) [reply src/main/java/com/singh/astha/notificationservice/config/AppConfig.java]: fixed

File: src/main/java/com/singh/astha/notificationservice/config/SecurityConfig.java (hunk 3 above)
Comment (astha-singh-2445) [reply src/main/java/com/singh/astha/notificationservice/config/SecurityConfig.java]: fixed

File: src/main/resources/application.properties (hunk 4 above)
Comment (astha-singh-2445) [reply src/main/resources/application.properties]: fixed

File: src/main/resources/application.properties [hunk 5]
```diff
@@ -3,4 +3,6 @@ spring.data.mongodb.auto-index-creation=true
 server.port=${SERVER_PORT:8081}
 spring.kafka.consumer.group-id=astha
 auth.token.public.key=${PUBLIC_KEY}
-spring.kafka.bootstrap-servers=${KAFKA_BOOTSTRAP_SERVERS}
\ No newline at end of file
+spring.kafka.bootstrap-servers=${KAFKA_BOOTSTRAP_SERVERS}
+springdoc.api-docs.path=/api-docs
+springdoc.swagger-ui.path=/swagger-ui-custom.html
```
Comment (ShivamNagpal) [inline src/main/resources/application.properties:8]: @astha-singh-2445 
nit: SpringDoc configure default values for these. Check if you actually need these values

**Review by ShivamNagpal — APPROVED** [review]

**Pull request merged**